docs/Actions-and-the-Dispatcher.th-TH.md

```diff
@@ -0,0 +1,43 @@
+---
+id: actions-and-the-dispatcher-th-TH
+title: Actions และ Dispatcher
+layout: docs
+category: Guides
+permalink: docs/actions-and-the-dispatcher-th-TH.html
+next: testing-flux-applications-th-TH
+lang: th-TH
+---
+
+บทความนี้ดั้งเดิมถูกเขียนไว้เป็น [post](http://facebook.github.io/react/blog/2014/07/30/flux-actions-and-the-dispatcher.html) ที่ [React blog](http://facebook.github.io/react/blog/) และได้ถูกแก้ไขเรียบเรียงและอัพเดทไว้ใหม่ที่นี่
+
+
+Dispatcher
+--------------
+
+Dispatcher มีเพียงหนึ่งเดียว (singleton) และทำงานเป็นเหมือนศูนย์กลางการเดินทางของข้อมูลทั้งหมดในแอพ Flux โดยที่ Dispatcher คือสำนักทะเบียนที่เก็บ callbacks ของ Stores เอาไว้ และสามารถเรียก callbacks เหล่านี้ได้ในลำดับที่ต้องการ _Store_ แต่ละตัวลงทะเบียน callback ไว้กับ dispatcher เมื่อมีข้อมูลใหม่เดินทางเข้ามาใน dispatcher มันจะใช้ callbacks เหล่านี้เพื่อกระจายข้อมูลนั้นไปยัง Stores ทุกๆ ตัว ซึ่งขั้นตอนของการเรียก callbacks เหล่านี้ถูกเริ่มต้นขึ้นจากคำสั่ง dispatch() ซึ่งรับข้อมูล payload เป็นเพียง argument เป็นหนึ่งเดียวเท่านั้น โดยทีปกติแล้ว่ข้อมูล payload นีสอดคล้องกับ _action_.
+
+<figure class="diagram">
+  <img src="/flux/img/flux-simple-f8-diagram-with-client-action-1300w.png" alt="data flow in Flux with data originating from user interactions" width=650 />
+</figure>
+
+
+Actions และ Action Creators
+---------------------------
+
+เมื่อมีข้อมูลใหม่เข้ามาในระบบ ไม่ว่าจะมาจาก interaction ของ user หรือจาก web api ที่เรียก ข้อมูลนั้นจะถูกทำให้เป็น _action_ — ซึ่งเป็นเพียง object ที่มีเพียง field ข้อมูลและประเภทของข้อมูลนั้น โดยที่ส่วนใหญ่แล้วเรามักจะสร้างชุดของคำสั่ง helper ที่เรียกว่า _action creators_ ที่ทำหน้าที่ทั้งสร้าง action object และยังส่งต่อ action นั้นไปยัง dispatcher อีกด้วย
+
+Action ที่แตกต่างกันจะมีประเภทของ action (type) ที่แตกต่างกัน ซึ่งเมื่อ Store รับค่า action เข้ามา มันจะตรวจสอบประเภทว่า action นี้เป็นประเภทใด และประมวลผลข้อมูลแยกตาม type ของ action ในแอพ Flux ทั้ง Stores และ Views ควบคุมตัวเองเป็นอิสระ โดยที่ object อื่นภายนอกไม่สามารถเข้ามายุ่งได้ Actions จะถูกส่งต่อเข้ามายัง Stores ด้วยวิธีการเรียกผ่าน callbacks ที่มันลงทะเบียนเอาไว้ก่อนหน้า ไม่ใช่โดย setter methods
```

Comment:
Can you please verify for me that this expresses the idea that the store's registered callbacks are getting used instead of setter methods?  Google translate is not helping me here, saying "registration took place earlier than the setter methods" which would be inaccurate.